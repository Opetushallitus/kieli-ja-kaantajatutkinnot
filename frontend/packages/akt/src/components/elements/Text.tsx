import { Typography, TypographyProps, TypographyVariant } from '@mui/material';

const variantInDiv = (variant: TypographyVariant, props: TypographyProps) => {
  const { children, ...rest } = props;

  return (
    <Typography variant={variant} {...rest}>
      {children}
    </Typography>
  );
};

export const H1 = (props: TypographyProps) => variantInDiv('h1', props);
export const H2 = (props: TypographyProps) => variantInDiv('h2', props);
export const H3 = (props: TypographyProps) => variantInDiv('h3', props);
export const H4 = (props: TypographyProps) => variantInDiv('h4', props);
export const H5 = (props: TypographyProps) => variantInDiv('h5', props);
export const H6 = (props: TypographyProps) => variantInDiv('h6', props);

export const Text = (props: TypographyProps) => variantInDiv('body1', props);
export const Caption = (props: TypographyProps) =>
  variantInDiv('caption', props);
