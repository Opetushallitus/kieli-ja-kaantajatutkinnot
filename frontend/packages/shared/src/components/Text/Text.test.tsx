import renderer from 'react-test-renderer';

import { Caption, H1, H2, H3, H4, H5, H6, Text } from './Text';

describe('Caption', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Caption>This is a caption</Caption>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('H1', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<H1>This is an H1</H1>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('H2', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<H2>This is an H2</H2>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('H3', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<H3>This is an H3</H3>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('H4', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<H4>This is an H4</H4>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('H5', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<H5>This is an H5</H5>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('H6', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<H6>This is an H6</H6>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Text', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Text>This is just text</Text>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
