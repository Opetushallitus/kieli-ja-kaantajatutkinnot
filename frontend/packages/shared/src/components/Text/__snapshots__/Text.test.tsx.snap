// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption should render correctly 1`] = `
<span
  className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
>
  This is a caption
</span>
`;

exports[`H1 should render correctly 1`] = `
<h1
  className="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
>
  This is an H1
</h1>
`;

exports[`H2 should render correctly 1`] = `
<h2
  className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
>
  This is an H2
</h2>
`;

exports[`H3 should render correctly 1`] = `
<h3
  className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
>
  This is an H3
</h3>
`;

exports[`H4 should render correctly 1`] = `
<h4
  className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
>
  This is an H4
</h4>
`;

exports[`H5 should render correctly 1`] = `
<h5
  className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
>
  This is an H5
</h5>
`;

exports[`H6 should render correctly 1`] = `
<h6
  className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
>
  This is an H6
</h6>
`;

exports[`Text should render correctly 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
>
  This is just text
</p>
`;
